### ISSUE TITLE
RadioButton is not vocalized as unselected by VoiceOver.

### ISSUE THREAD

**Issue by aferret0** [flutter/flutter#170422]
### Steps to reproduce

Using iOS 18.5
 
1. Navigate to an unselected RadioButton 

### Expected results

1. Navigate to an unselected RadioButton
2. VoiceOver vocalizes the RadioButton state

Since Android (Talkback) does it, I would expect iOS to behave the same way — but maybe I'm wrong.

### Actual results

1. Navigate to an unselected RadioButton
2. VoiceOver does not announce the RadioButton state

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: SafeArea(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              const Text('Select an option:'),
              const SizedBox(height: 20),
              const MyRadioButtons(),
            ],
          ),
        ),
      ),
    );
  }
}

class MyRadioButtons extends StatefulWidget {
  const MyRadioButtons({super.key});

  @override
  State<MyRadioButtons> createState() => _MyRadioButtonsState();
}

class _MyRadioButtonsState extends State<MyRadioButtons> {
  String? _selectedValue;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        RadioListTile<String>(
          title: const Text('Option 1'),
          value: 'option1',
          groupValue: _selectedValue,
          onChanged: (String? value) {
            setState(() {
              _selectedValue = value;
            });
          },
        ),
        RadioListTile<String>(
          title: const Text('Option 2'),
          value: 'option2',
          groupValue: _selectedValue,
          onChanged: (String? value) {
            setState(() {
              _selectedValue = value;
            });
          },
        ),
        Text('Selected: ${_selectedValue ?? "None"}'),
      ],
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.29.2, on macOS 15.5 24F74 darwin-arm64, locale fr-FR) [12,2s]
    • Flutter version 3.29.2 on channel stable at /Users/34080J/fvm/versions/3.29.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c236373904 (3 months ago), 2025-03-13 16:17:06 -0400
    • Engine revision 18b71d647a
    • Dart version 3.7.2
    • DevTools version 2.42.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1) [4,7s]
    • Android SDK at /Users/34080J/Library/Android/sdk
    • Platform android-35, build-tools 35.0.1
    • ANDROID_HOME = /Users/34080J/Library/Android/sdk
    • Java binary at: /Library/Java/JavaVirtualMachines/openlogic-openjdk-17.jdk/Contents/Home/bin/java
      This JDK is specified in your Flutter configuration.
      To change the current JDK, run: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment OpenLogic-OpenJDK (build 17.0.14+7-adhoc.admin.jdk17u)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.4) [1 876ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16F6
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [27ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2) [27ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.5+-12932927-b750.29)

[✓] IntelliJ IDEA Ultimate Edition (version 2025.1) [26ms]
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.98.2) [8ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.108.0

[✓] Connected device (5 available) [6,1s]
    • sdk gphone64 arm64 (mobile)     • emulator-5554             • android-arm64  • Android 15 (API 35) (emulator)
    • iPhone d’Alexandre (mobile)     • 00008140-001539621E51801C • ios            • iOS 18.5 22F76
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 15.5 24F74 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 15.5 24F74 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 137.0.7151.69

[✓] Network resources [243ms]
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by tirth-patel-nc** [comment]
# Triage Report

I was able to reproduce the issue on iOS using the provided code sample on both the `stable (3.32.2)` and `master (3.33.0-1.0.pre.474)` channels. #125499 likely added the semantics label for the selected state, but the issue being reported is specifically about the unselected state.

```
selected option 1 button
option 2 button <-- this is not selected which per OP should be announced by as unselected
```

On Android, it does announce the unselected state as expected.

**Comment by chunhtai** [comment]
Radio button is treated as a UIswitch in iOS, not sure why it doesn't work. will need more investigation

**Comment by chunhtai** [comment]
issue with the same root cause https://github.com/flutter/flutter/issues/146987
cc @hannah-hyj
